targets/psa_switch/psa_switch.h

```diff
@@ -119,6 +127,50 @@ class PsaSwitch : public Switch {
 
   void set_transmit_fn(TransmitFn fn);
 
+  // overriden interfaces
+  Counter::CounterErrorCode
+  read_counters(cxt_id_t cxt_id,
+                const std::string &counter_name,
+                size_t index,
+                MatchTableAbstract::counter_value_t *bytes,
+                MatchTableAbstract::counter_value_t *packets) override {
+    Context *context = get_context(cxt_id);
+    ExternType *ex = context->get_extern_instance(counter_name).get();
+    if (!ex) return Counter::CounterErrorCode::INVALID_COUNTER_NAME;
+    PSA_Counter *counter = static_cast<PSA_Counter*>(ex);
+    if (index >= counter->size()) 
+      return Counter::CounterErrorCode::INVALID_INDEX;
+    counter->get_counter(index).query_counter(bytes, packets);
+    return Counter::CounterErrorCode::SUCCESS;
+  }
+
+  Counter::CounterErrorCode
+  write_counters(cxt_id_t cxt_id,
+                 const std::string &counter_name,
+                 size_t index,
+                 MatchTableAbstract::counter_value_t bytes,
+                 MatchTableAbstract::counter_value_t packets) override {
+    Context *context = get_context(cxt_id);
+    ExternType *ex = context->get_extern_instance(counter_name).get();
+    if (!ex) return Counter::CounterErrorCode::INVALID_COUNTER_NAME;
+    PSA_Counter *counter = static_cast<PSA_Counter*>(ex);
+    if (index >= counter->size())
+      return Counter::CounterErrorCode::INVALID_INDEX;
+    counter->get_counter(index).write_counter(bytes, packets);
+    return Counter::CounterErrorCode::SUCCESS;
+  }
+
+  Counter::CounterErrorCode
+  reset_counters(cxt_id_t cxt_id,
+                 const std::string &counter_name) override {
+    Context *context = get_context(cxt_id);
+    ExternType *ex = context->get_extern_instance(counter_name).get();
+    if (!ex) return Counter::CounterErrorCode::INVALID_COUNTER_NAME;
+    PSA_Counter *counter = static_cast<PSA_Counter*>(ex);
+    return counter->reset_counters();
+  }
+
```

Comment:
This may wait until after this PR is merged, but I would recommend to urgently override all the `RuntimeInterface` methods (https://github.com/p4lang/behavioral-model/blob/master/include/bm/bm_sim/runtime_interface.h) which are not yet supported by PSASwitch and log an error message + return an error code.